packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts

```diff
@@ -1055,6 +1125,49 @@ export class Cluster extends ClusterBase {
       ],
     });
 
+    const autoModeEnabled = !(props.defaultCapacityType !== undefined && props.defaultCapacityType !== DefaultCapacityType.AUTOMODE);
+
+    // Throw when using nodePools or nodeRole without using AUTOMODE
+    if (!autoModeEnabled && (props.compute?.nodePools || props.compute?.nodeRole)) {
+      throw new Error('Cannot specify nodePools or nodeRole without using DefaultCapacityType.AUTOMODE');
+    }
+
+    // When using AUTOMODE, defaultCapacity and defaultCapacityInstance cannot be specified
+    if (autoModeEnabled && (props.defaultCapacity !== undefined || props.defaultCapacityInstance !== undefined)) {
+      throw new Error('Cannot specify defaultCapacity or defaultCapacityInstance when using Auto Mode. Auto Mode manages compute resources automatically.');
+    }
+
+    // Node pool values are case-sensitive and must be general-purpose and/or system
+    if (props.compute?.nodePools) {
+      const validNodePools = ['general-purpose', 'system'];
+      const invalidPools = props.compute.nodePools.filter(pool => !validNodePools.includes(pool));
+      if (invalidPools.length > 0) {
+        throw new Error(`Invalid node pool values: ${invalidPools.join(', ')}. Valid values are: ${validNodePools.join(', ')}`);
+      }
+    }
```

Comment:
We can move these to `if (autoModeEnabled)` below

Reply:
DONE